lib/kernel/src/logger_olp.erl

```diff
@@ -507,6 +507,24 @@ check_load(State = #{id:=_Name, mode_ref := ModeRef, mode := Mode,
      ?update_other(flushes,FLUSHES,_NewFlushes,
                    State4#{last_qlen => QLen})}.
 
+%% Calling process_info(self(), memory) is linear on the size of the message queue,
+%% which is an extremely bad thing to do in high load situations, so only do that
+%% if the value actually is required.
+-ifdef(OBSERVER_MOD). % does ?observe/2 use Mem
+maybe_self_memory(_State) -> do_self_memory().
+-else.
+-ifdef(SAVE_STATS). % does ?update_max_mem/2 use Mem
```

Comment:
These `ifdef`s are needed because of pre-existing preprocessor usage in `logger_olp`.